shared/naturalcrit/markdown.js

```diff
@@ -206,6 +206,38 @@ const mustacheInjectBlock = {
 	}
 };
 
+const superSubScripts = {
+	name  : 'superSubScripts',
+	level : 'inline',
+	start(src) { return src.match(/.*\^\^(.+)\^\^/)?.index; },  // Hint to Marked.js to stop and check for a match
```

Comment:
This regex can be made faster by removing the starting `.*` (it doesn't matter what is before the beginning of the superscript) and making sure the central `.+` excludes newlines (so this isn't madly searching the entire document each time it sees a ^`.

I suggest we also limit the regex so the super/sub must not start with whitespace. i.e. `^some text^` is valid for superscript, but `^ some text^`  and `^some text ^` are not. This would help avoid conflicts with normal use cases of `^` and also match the existing markdown specs for `*italics*`.

Reply:
This should be resolved by changing the  (.+) to ([^\s\^][^\^]*[^\s\^]).